internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/issuecreator/JIRAIssueCreator.java

```diff
@@ -0,0 +1,242 @@
+/*
+ *
+ *   Copyright (c) 2018, WSO2 Inc., WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ * /
+ */
+
+package org.wso2.security.tools.scanner.dependency.js.issuecreator;
+
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.security.tools.scanner.dependency.js.constants.IssueCreatorConstants;
+import org.wso2.security.tools.scanner.dependency.js.constants.JSScannerConstants;
+import org.wso2.security.tools.scanner.dependency.js.exception.ConfigParserException;
+import org.wso2.security.tools.scanner.dependency.js.exception.IssueCreatorException;
+import org.wso2.security.tools.scanner.dependency.js.utils.ConfigParser;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.naming.AuthenticationException;
+
+
+/**
+ * Jira Issue Creator
+ */
+public class JIRAIssueCreator extends IssueCreator {
+    private static final Logger log = Logger.getLogger(JIRAIssueCreator.class);
+    private JIRARestClient jira;
+    private static String ticketSubject;
+    private static String projectKey;
+    private static String issueLabel;
+    private static String issueType;
+    private static final Charset UTF_8 = StandardCharsets.UTF_8;
+
+    public JIRAIssueCreator(JIRARestClient jira, char[] username, char[] password) {
+        super(username, password);
+        this.jira = jira;
+    }
+
+
+    public static void setTicketSubject(String ticketSubject) {
+        JIRAIssueCreator.ticketSubject = ticketSubject;
+    }
+
+
+    public static void setProjectKey(String projectKey) {
+        JIRAIssueCreator.projectKey = projectKey;
+    }
+
+
+    public static void setIssueLabel(String issueLabel) {
+        JIRAIssueCreator.issueLabel = issueLabel;
+    }
+
+
+    public static void setIssueType(String issueType) {
+        JIRAIssueCreator.issueType = issueType;
+    }
+
+    @Override
+    public void handleIssueCreatorAPICall(HashMap<String, String> responseMapper, HashMap<String, String> fileMapper)
```

Comment:
Missing javadoc comment. You should either explain properly all the input parameters and what is done in this overridden method, or should inherit from the super class' doc using {@inheritDoc} (Super class doc needs to be in proper format and comprehensive)